TYPO3.Flow/Classes/TYPO3/Flow/Utility/Environment.php

```diff
@@ -58,28 +47,22 @@ public function __construct(\TYPO3\Flow\Core\ApplicationContext $context)
      *
      * @param string $temporaryDirectoryBase Base path of the temporary directory, with trailing slash
      * @return void
+     * @throws Exception
+     * @deprecated since 2.3 - Set the environment variable FLOW_PATH_TEMPORARY_BASE to change the temporary directory base, see Bootstrap::defineConstants()
      */
     public function setTemporaryDirectoryBase($temporaryDirectoryBase)
     {
-        $this->temporaryDirectoryBase = $temporaryDirectoryBase;
-        $this->temporaryDirectory = null;
+        throw new Exception('Changing the temporary directory path during runtime is no longer supported. Set the environment variable FLOW_PATH_TEMPORARY_BASE to change the temporary directory base', 1441355116);
     }
 
     /**
      * Returns the full path to Flow's temporary directory.
      *
-     * @return string Path to PHP's temporary directory
-     * @api
```

Comment:
Yes, it's totally doable. But this is meant to be a fix for 3.0.1 (and 2.3.x+1) mainly with little as possible impact.
So I'd suggest to re-add the api annotation as you suggested, get this one merged as patchlevel release and then have `Environment` deprecated for 3.1..
BTW: Is it possible to target a PR for 2.3.x+1 but *not* for 3.0.x+1?